src/RNSCommission.sol

```diff
@@ -0,0 +1,129 @@
+        virtual
+        returns (
+            address _skyMavisTreasuryAddr,
+            address _roninNetworkTreasuryAddr
+        )
+    {
+        return (_skyMavisTreasury, _roninNetworkTreasury);
+    }
+    function setTreasuryAddresses(
+        address payable _skyMavisTreasuryAddr,
+        address payable _roninNetworkTreasuryAddr
+    ) external virtual onlyRole(DEFAULT_ADMIN_ROLE) {
+        _setTreasuryAddresses(_skyMavisTreasuryAddr, _roninNetworkTreasuryAddr);
+    }
+
+    function getCommissionSchema()
+        public
+        view
+        virtual
+        returns (CommissionSchema memory)
+    {
+        return _commissionsSchema;
+    }
+
+    function _setTreasuryAddresses(
+        address payable _skyMavisTreasuryAddr,
+        address payable _roninNetworkTreasuryAddr
+    ) internal virtual {
+        _skyMavisTreasury = _skyMavisTreasuryAddr;
+        _roninNetworkTreasury = _roninNetworkTreasuryAddr;
+    }
+
+    function setCommissionSchema(
+        uint256 _skyMavisRatio,
+        uint256 _roninRatio,
+        Commission[] calldata _otherCommissions
+    ) external onlyRole(OPERATOR_ROLE) {
+        uint256 _sum = _skyMavisRatio + _roninRatio;
+        _commissionsSchema.skyMavisRatio = _skyMavisRatio;
+        _commissionsSchema.roninRatio = _roninRatio;
+        delete _commissionsSchema.others;
+        for (uint i = 0; i < _otherCommissions.length; i++) {
+            _sum += _otherCommissions[i].ratio;
+            _commissionsSchema.others.push(_otherCommissions[i]);
+        }
+        require(_sum == MAX_PERCENTAGE, "RNSCommission: invalid ratio");
+        _commissionsSchema.sumRatios = _sum;
+    }
+
+    function getAllocations(
+        uint256 _total
+    ) external view returns (Allocation[] memory _allocs) {
+        CommissionSchema memory _schema = getCommissionSchema();
+
+        // First 2 slots is for Sky Mavis + Ronin.
+        uint256 _allocNum = 2 + _schema.others.length;
+        _allocs = new Allocation[](_allocNum);
+        _allocs[0] = Allocation({
+            recipient: _skyMavisTreasury,
+            ratio: _schema.skyMavisRatio,
+            value: _computePercentage(_total, _schema.skyMavisRatio)
+        });
+
+        _allocs[1] = Allocation({
+            recipient: _roninNetworkTreasury,
+            ratio: _schema.roninRatio,
+            value: _computePercentage(_total, _schema.roninRatio)
+        });
+
+        for (uint256 i; i < _schema.others.length; i++) {
+            address payable _owner = _schema.others[i].owner;
+            uint256 _ratio = _schema.others[i].ratio;
+            _allocs[2 + i] = Allocation({
+                recipient: _owner,
+                ratio: _ratio,
+                value: _computePercentage(_total, _ratio)
+            });
+        }
+    }
+    function _computePercentage(
+        uint256 value,
+        uint256 percentage
+    ) internal pure virtual returns (uint256) {
+        return (value * percentage) / MAX_PERCENTAGE;
```

Comment:
Use Math.mulDiv of OZ for avoiding some arithmetic issues, such as overflow in multiplication or divide by 0